libs/framework/src/bundle_archive.c

```diff
@@ -28,644 +28,415 @@
-
-	return status;
-}
-
 celix_status_t bundleArchive_getId(bundle_archive_pt archive, long *id) {
-	celix_status_t status = CELIX_SUCCESS;
-
-	if (archive->id < 0) {
-		FILE *bundleIdFile;
-		char id[256];
-		char bundleId[512];
-		snprintf(bundleId, sizeof(bundleId), "%s/bundle.id", archive->archiveRoot);
-
-		bundleIdFile = fopen(bundleId, "r");
-		if(bundleIdFile!=NULL){
-			fgets(id, sizeof(id), bundleIdFile);
-			fclose(bundleIdFile);
-			sscanf(id, "%ld", &archive->id);
-		}
-		else{
-			status = CELIX_FILE_IO_EXCEPTION;
-		}
-	}
-
-	if (status == CELIX_SUCCESS) {
-		*id = archive->id;
-	}
+     *id = archive->id;
+	return CELIX_SUCCESS;
+}
 
-	framework_logIfError(archive->fw->logger, status, NULL, "Could not get archive id");
+long celix_bundleArchive_getId(bundle_archive_pt archive) {
+    return archive->id;
+}
 
-	return status;
+const char* celix_bundleArchive_getSymbolicName(bundle_archive_pt archive) {
+    return archive->bundleSymbolicName;
 }
 
 celix_status_t bundleArchive_getLocation(bundle_archive_pt archive, const char **location) {
-	celix_status_t status = CELIX_SUCCESS;
-	if (archive->location == NULL) {
-		FILE *bundleLocationFile;
-		char bundleLocation[512];
-		char loc[256];
-
-		snprintf(bundleLocation, sizeof(bundleLocation), "%s/bundle.location", archive->archiveRoot);
-
-		bundleLocationFile = fopen(bundleLocation, "r");
-		if(bundleLocationFile!=NULL){
-			fgets(loc, sizeof(loc), bundleLocationFile);
-			fclose(bundleLocationFile);
-			archive->location = strdup(loc);
-		}
-		else{
-			status = CELIX_FILE_IO_EXCEPTION;
-		}
-	}
-
-	if (status == CELIX_SUCCESS) {
-		*location = archive->location;
-	}
-
-	framework_logIfError(archive->fw->logger, status, NULL, "Could not get archive location");
-
-	return status;
+    *location = archive->location;
+    return CELIX_SUCCESS;
 }
 
 celix_status_t bundleArchive_getArchiveRoot(bundle_archive_pt archive, const char **archiveRoot) {
-	*archiveRoot = archive->archiveRoot;
-	return CELIX_SUCCESS;
+    *archiveRoot = archive->archiveRoot;
+    return CELIX_SUCCESS;
 }
 
 celix_status_t bundleArchive_getCurrentRevisionNumber(bundle_archive_pt archive, long *revisionNumber) {
-	celix_status_t status = CELIX_SUCCESS;
-	bundle_revision_pt revision;
-	*revisionNumber = -1;
-
-	status = CELIX_DO_IF(status, bundleArchive_getCurrentRevision(archive, &revision));
-	status = CELIX_DO_IF(status, bundleRevision_getNumber(revision, revisionNumber));
-
-	framework_logIfError(archive->fw->logger, status, NULL, "Could not get current revision number");
-
-	return status;
+    *revisionNumber = 1; //NOTE bundle revision is deprecated
+    return CELIX_SUCCESS;
 }
 
 celix_status_t bundleArchive_getCurrentRevision(bundle_archive_pt archive, bundle_revision_pt *revision) {
-	*revision = linkedList_isEmpty(archive->revisions) ? NULL : linkedList_getLast(archive->revisions);
-	return CELIX_SUCCESS;
+    *revision = archive->revision;
```

Comment:
`bundleArchive_revise` changes `archive->revision`. Thrad safety issue.

Reply:
It's more difficult to fix this use-after-free without changing interface.
Consider when it returns, `archive->revision` could be freed by `bundleArchive_revise` but is stilled used by the caller.